src/d/actor/d_a_obj_tower.cpp

```diff
@@ -4,50 +4,164 @@
 //
 
 #include "d/actor/d_a_obj_tower.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_procname.h"
+#include "dolphin/mtx/mtx.h"
 #include "dolphin/types.h"
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_printf.h"
 
 /* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* i_this) {
+    return ((daObjTower_c*)i_this)->CreateHeap();
 }
 
 /* 00000098-0000020C       .text CreateHeap__12daObjTower_cFv */
-void daObjTower_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObjTower_c::CreateHeap() {
+    J3DModelData* modelData = (J3DModelData*)dComIfG_getObjectRes("X_tower", 4);
+
+    JUT_ASSERT(0x56, modelData != 0);
+
+    mpModel = mDoExt_J3DModel__create(modelData, 0, 0x11020203);
+
+    if (mpModel == NULL)
+        return FALSE;
+
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(shape_angle.y);
+    mDoMtx_stack_c::scaleM(mScale.x, mScale.y, mScale.z);
+    mDoMtx_copy(mDoMtx_stack_c::get(), mMtx);
+
+    mpBgW = new dBgW();
+
+    if (mpBgW == NULL)
+        return FALSE;
+
+    cBgD_t* pData = (cBgD_t*)dComIfG_getObjectRes("X_tower", 7);
+
+    return mpBgW->Set(pData, cBgW::MOVE_BG_e, &mMtx) ? FALSE : TRUE;
 }
 
 /* 0000020C-0000028C       .text CreateInit__12daObjTower_cFv */
 void daObjTower_c::CreateInit() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    fopAcM_setCullSizeBox(this, -30000.0, -5000.0, -30000.0, 30000.0, 40000.0, 30000.0);
+    dComIfG_Bgsp()->Regist(mpBgW, this);
+    field_0x2d0 = 1;
+    set_mtx();
 }
 
 /* 0000028C-0000030C       .text set_mtx__12daObjTower_cFv */
 void daObjTower_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(mScale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mDoMtx_copy(mDoMtx_stack_c::get(), mpModel->mBaseTransformMtx);
 }
 
+cPhs__Step daObjTower_c::_create() {
+    cPhs__Step PVar3;
+
+    fopAcM_SetupActor(this, daObjTower_c);
+
+    this->field_0x2d0 = 0;
```

Comment:
```suggestion
    field_0x2d0 = 0;
```
you can remove the `this->` from all these C++-style member functions. only `i_this->` is needed in C-style functions.